### PR TITLE
Installation notes for ubunutu from git

### PR DESCRIPTION
**Pull request by jmikedupont2**
**Notes on Installation on Ubuntu 22.04.1 LTS with pyenv 3.10**

Hello everyone,

I wanted to share my experience with the installation process on Ubuntu 22.04.1 LTS using pyenv 3.10. Please note that these are preliminary notes and not yet ready for merging. The purpose of this message is to document the issues I encountered during the installation process.

**Description**

I installed the Azure CLI on Ubuntu 22.04.1 LTS using pyenv 3.10 as the Python version. However, I ran into a few challenges during the installation process. These challenges include [briefly describe the issues you encountered].

**Testing Guide**

I followed the steps outlined in the [Azure CLI installation guide](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli) for Ubuntu and used pyenv to manage the Python version. Despite following the instructions, I encountered the aforementioned issues.

**Summary of Azure CLI Installation Experience**

During my attempt to install and configure the Azure CLI, I went through several methods. Here's a detailed breakdown of each step:

1. **Cloning the Repository:**
   - I started by cloning the Azure CLI repository from GitHub.
   - Then, I navigated into the cloned `azure-cli` directory.

3. **Installer Script Challenges:**
   - I encountered difficulties while trying to use the installer script.
   - After downloading the Azure CLI installer script using `curl`, I used the `bash` command to execute it.

   **Issues Encountered:**
   - The installer script failed multiple times.
   - The Azure CLI was not compatible with Python 3.13.
   - The `requirements.txt` file, which is usually used for dependencies, was named `requires.txt`. This naming mismatch led to further challenges.

5. **Downloading Installer and Python Scripts:**
   - I followed the steps to download the installer script (`InstallAzureCli.sh`) and the Python script (`install.py`).

7. **Running the Python Installer Script:**
   - To proceed, I executed the Python installer script. This involved individually installing each required package.

   **Centralized Package Installation:**
   - It would greatly enhance the process to have a centralized package that can install all three components as sub-projects.

8. **Verifying Azure CLI Installation:**
   - To ensure the installation was successful, I tested the Azure CLI by using commands such as `az` and `az login`.
   - I also sourced the `.bash_profile` to update my environment settings.

9. **Managing Requirements:**
   - I managed to resolve some issues by merging the requirements from all the sub-modules.
   - This allowed me to successfully install the required dependencies using `pip`.

10. **Logging In Using Device Code:**
    - Finally, I logged in using the Azure CLI's device code authentication method.

---

Feel free to use this revised version or make any further adjustments as needed. The goal is to ensure clarity and effective communication of your experience and suggestions.
---

Please keep in mind that these notes are preliminary and not ready for merging. I'm sharing this information to raise awareness of the challenges I faced during the installation process. If you have any insights or suggestions on how to resolve these issues, I would greatly appreciate your input.

Thank you for your attention, and I look forward to collaborating to improve the installation process.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: requirements.txt
Comment (jiasli) [inline requirements.txt:1]: This file is only used for `azdev setup -c` from https://github.com/Azure/azure-cli-dev-tools

```
> azdev setup -c --debug
...
Installing `requirements.txt`...
cli.azdev.utilities.command: Running: D:\cli\py310\Scripts\python -m pip install -r D:\cli\azure-cli\requirements.txt
```

It is not meant to manage Azure CLI's dependencies.

File: install.py
Comment (jiasli) [inline install.py:1]: This file is merely a copy of `scripts/curl_install_pypi/install.py`. Why is it copied to the root folder?

File: README.md [hunk 1]
```diff
@@ -221,3 +221,26 @@ For more information see the [Code of Conduct FAQ](https://opensource.microsoft.
 
 If you would like to become an active contributor to this project please
 follow the instructions provided in [Microsoft Open Source Guidelines](https://opensource.microsoft.com/collaborate).
+
+
+## Install
+
+  620  git clone https://github.com/Azure/azure-cli
+  621  cd azure-cli/
+  704  ~/lib/azure-cli/bin/az
```
Comment (jiasli) [inline README.md:230]: What are these number before commands?

File: InstallAzureCli.sh
Comment (jiasli) [inline InstallAzureCli.sh:1]: This is a copy of `scripts/curl_install_pypi/install`.

File: package-lock.json
Comment (jiasli) [inline package-lock.json:1]: `package-lock.json` is used by `npm`: https://docs.npmjs.com/cli/v9/configuring-npm/package-lock-json. Why is it included here?

File: providers.txt
Comment (jiasli) [inline providers.txt:1]: This file is generated by 

```
754 az provider list --query "[].resourceTypes[].resourceType" --output table > providers.txt
```

and shouldn't be checked in.

**Comment by jiasli** [comment]
Thanks for the contribution. However, installing Azure CLI from Git source code is not and will not be supported.

An similar alternative is to follow https://github.com/Azure/azure-cli/issues/20476 to install from PyPI but this is not officially supported either.

Supported packages are documented by https://learn.microsoft.com/en-us/cli/azure/install-azure-cli.

Thank you for your understanding.

**Comment by jmikedupont2** [comment]
Hey there,

I completely understand and respect your decision to close it, and I want
to make it clear that I never expected them to merge it. My intention was
simply to document the process for anyone else who might face the same
challenges. The numbers I provided are directly from my bash history and
represent the commands I executed to build and install from the git
repository. I wanted to share this information to help others who might be
interested in exploring this unsupported approach in the spirit of freedom.
Please know that I had no intention of being rude or disrespectful in any
way. Thank you for considering my perspective.

Best regards,
mike

**Comment by jmikedupont2** [comment]
How does this sound for inclusion in the readme so we can help people
direct people to the fact that installation from get is not supported
because I think that would be important to note very in the beginning on
the installation notes cuz I had to look for a long time and I read through
everything I read through the source that I needed to get it installed.

"We would like to draw attention to the fact that while there are
historical instances of users attempting to install the software directly
from the git repository, we want to emphasize that this approach is not
officially supported. It's worth noting that users might encounter
references to closed pull requests and discussions that document such
attempts. However, we recommend following the officially recommended
installation methods to ensure a smoother experience with the software."
would you agree if I created a post request that would add that to the
installation segment of the readme so we can spare people the search

**Comment by jiasli** [comment]
@jmikedupont2, Our `README.md` does have a pointer to the official installation document: https://github.com/Azure/azure-cli#installation. It also contains instruction for developers to configure their machine for development: https://github.com/Azure/azure-cli#developer-setup.

We only mention supported things. There are so many _unsupported_ things that we can't exhaust, so I would prefer to keep the document as simple as possible. Thanks for your understanding.

**Comment by jmikedupont2** [comment]
But I find that you wasted an hour of my life and I'm trying to compensate for that because other people's time will be wasted as well

**Pull request closed without merging**